modmail/utils/pagination.py

```diff
@@ -0,0 +1,281 @@
+        paginator._embed.set_footer(text=paginator.get_footer())
+        # if there's only one page, don't send the view
+        if len(paginator.pages) < 2:
+            await channel.send(embeds=[paginator._embed])
+            return
+        else:
+            msg: discord.Message = await channel.send(embeds=[paginator._embed], view=paginator)
+
+        await paginator.wait()
+        await msg.edit(view=None)
+
+    async def interaction_check(self, interaction: Interaction) -> bool:
+        """Check if the interaction is by the author of the paginator."""
+        if self.only_users is not None:
+            logger.trace(f"All allowed users: {self.only_users}")
+            if any(user.id == interaction.user.id for user in self.only_users):
+                logger.debug("User is in allowed users")
+                return True
+        if self.only_roles is not None:
+            logger.trace(f"All allowed roles: {self.only_roles}")
+            user_roles = [role.id for role in interaction.user.roles]
+            for role in self.only_roles:
+                if role.id in user_roles:
+                    logger.debug("User is in allowed roles")
+                    return True
+        await interaction.response.send_message(
+            content="You are not authorised to use this paginator.", ephemeral=True
+        )
+        return False
+
+    def get_footer(self) -> str:
+        """Returns the footer text."""
+        self._embed.description = self._pages[self._index]
+        page_indicator = f"Page {self._index+1}/{len(self._pages)}"
+        footer_txt = (
+            f"{self.footer_text} ({page_indicator})" if self.footer_text is not None else page_indicator
+        )
+        return footer_txt
+
+    def modify_states(self) -> None:
+        """
+        Disable specific components depending on paginator page and length.
+
+        If the paginatot has less than two pages, the jump buttons will be disabled.
+        If the paginator is on the first page, the jump first/move back buttons will be disabled.
+        if the paginator is on the last page, the jump last/move forward buttons will be disabled.
+        """
+        less_than_2_pages = len(self._pages) <= 2
+        components = {
+            "pag_jump_first": less_than_2_pages,
+            "pag_back": False,
+            "pag_next": False,
+            "pag_jump_last": less_than_2_pages,
+        }
+
+        if self._index == 0:
+            components["pag_jump_first"] = True
+            components["pag_back"] = True
+
+        if self._index == len(self._pages) - 1:
+            components["pag_next"] = True
+            components["pag_jump_last"] = True
+
+        for child in self.children:
+            if child.custom_id in components.keys():
+                if getattr(child, "disabled", None) is not None:
+                    child.disabled = components[child.custom_id]
+
+    async def send_page(self, interaction: Interaction) -> None:
```

Comment:
But that doesn't make it clear that we are changing what the user sees, does it?

Reply:
Maybe... but send_page is misleading because we're not sending anything, we're updating something. Or editing. Maybe "edit_page".. or "update_discord" or something..